homeassistant/components/frontend/__init__.py

```diff
@@ -533,6 +533,7 @@ async def resolve(
         """
         if (
             request.path != "/"
+            and len(request.url.parts) > 1
```

Comment:
We should add a test.

Reply:
I dropped testing this because I couldn't figure out how to correctly handle it. I had a test like the following:
```
async def test_handle_troublesome_urls(mock_http_client):
    """Test that troublesome URLs like // are handled gracefully."""
    resp = await mock_http_client.get("//")
    assert resp.status == 200
```
The above passes, but to verify the validity of the test, I removed my updated parts length check, and it still passed...  In the browser, without the check I see a 500 response code, so I was expecting to see the same in the tests.